assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -1229,7 +1230,8 @@ public ActionReturn setGrade(Map<String, Object> params) {
         if (submission != null) {
             boolean anonymousGrading = assignmentService.assignmentUsesAnonymousGrading(assignment);
             try {
-                return new ActionReturn(submissionToMap(activeSubmitters, assignment, new SimpleAssignment(assignment),  submission, true));
+                return new ActionReturn(Map.of("assignmentCloseTime", assignment.getCloseDate(),
```

Comment:
I'm not actually sure. I think maybe the data being returned was just too verbose for a saved grade. I will check my reasoning there.

Reply:
It is needed. I needed to get assignmentCloseTime returned so it could be updated on the current submission object. However, I may rethink that a little. I think we don't need all that data sending back on the network.

Reply:
No, it's needed as it is. Previously, the assignmentCloseTime was being serialised with each submission and bloating the json unnecessarily, so I moved it out of the submissionMap.

Reply:
Allright!